catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -92,39 +123,193 @@ public void initialize(Map<String, String> config, CatalogInfo info) throws Runt
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof TopicExistsException) {
+        throw new TopicAlreadyExistsException(e, "Topic %s already exists", ident);
+
+      } else if (e.getCause() instanceof InvalidReplicationFactorException) {
+        throw new IllegalArgumentException(
+            "Invalid replication factor for topic " + ident + e.getCause().getMessage(), e);
+
+      } else if (e.getCause() instanceof InvalidConfigurationException) {
+        throw new IllegalArgumentException(
+            "Invalid properties for topic " + ident + e.getCause().getMessage(), e);
+
+      } else {
+        throw new RuntimeException("Failed to create topic in Kafka" + ident, e);
+      }
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Failed to create topic in Kafka" + ident, e);
+    }
+
+    return KafkaTopic.builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withProperties(properties)
+        .withAuditInfo(
+            AuditInfo.builder()
+                .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                .withCreateTime(Instant.now())
+                .build())
+        .build();
   }
 
   @Override
   public Topic alterTopic(NameIdentifier ident, TopicChange... changes)
       throws NoSuchTopicException, IllegalArgumentException {
-    throw new UnsupportedOperationException();
+    KafkaTopic topic = (KafkaTopic) loadTopic(ident);
+    String newComment = topic.comment();
+    int oldPartitionCount = Integer.parseInt(topic.properties().get(PARTITION_COUNT));
+    int newPartitionCount = oldPartitionCount;
+    Map<String, String> alteredProperties = Maps.newHashMap(topic.properties());
+    List<AlterConfigOp> alterConfigOps = Lists.newArrayList();
+    for (TopicChange change : changes) {
+      if (change instanceof TopicChange.UpdateTopicComment) {
+        newComment = ((TopicChange.UpdateTopicComment) change).getNewComment();
+
+      } else if (change instanceof TopicChange.SetProperty) {
+        TopicChange.SetProperty setProperty = (TopicChange.SetProperty) change;
+        if (PARTITION_COUNT.equals(setProperty.getProperty())) {
+          // alter partition count
+          newPartitionCount = setPartitionCount(setProperty, newPartitionCount, alteredProperties);
+        } else {
+          // alter other properties
+          setProperty(setProperty, alteredProperties, alterConfigOps);
+        }
+
+      } else if (change instanceof TopicChange.RemoveProperty) {
+        removeProperty((TopicChange.RemoveProperty) change, alteredProperties, alterConfigOps);
+
+      } else {
+        throw new IllegalArgumentException("Unsupported topic change: " + change);
+      }
+    }
+
+    if (newPartitionCount != oldPartitionCount) {
+      doPartitionCountIncrement(ident.name(), newPartitionCount);
+    }
+
+    if (!alterConfigOps.isEmpty()) {
+      doAlterTopicConfig(ident.name(), alterConfigOps);
+    }
```

Comment:
Can we use one method or operation to implement it, I am afraid that the `alter` API may have partial success and failures.

Reply:
There may indeed be cases of partial success, as we have not committed to the transactional nature of this method externally. This is to align with the behavior of [Kafka's alter interface](https://github.com/apache/kafka/blob/4cb6806cb8f1cdbf9f47cb6521127fd3f49fa712/clients/src/main/java/org/apache/kafka/clients/admin/Admin.java#L528-L529).